File: typescript/test/com/murex/tbw/storage/InMemoryRepository.ts
import { Repository } from "../../../../../src/com/murex/tbw/storage/Repository";
import { Invoice } from "../../../../../src/com/murex/tbw/purchase/Invoice";
import * as Immutable from "immutable";

export class InMemoryRepository implements Repository {
  private readonly invoices = new Map<number, Invoice>();

  addInvoice(invoice: Invoice): void {
    this.invoices.set(invoice.id, invoice);
  }

  getInvoiceMap(): Immutable.Map<number, Invoice> {
    return Immutable.Map<number, Invoice>(this.invoices);
  }
}

File: scala/src/test/scala/com/murex/tbw/storage/InMemoryRepository.scala
package com.murex.tbw.storage

import com.murex.tbw.purchase.Invoice

class InMemoryRepository extends Repository {
  private val invoiceMap: scala.collection.mutable.Map[Int, Invoice] =
    scala.collection.mutable.Map[Int, Invoice]()

  override def addInvoice(invoice: Invoice): Unit = invoiceMap(invoice.id) = invoice

  override def getInvoiceMap: Map[Int, Invoice] = invoiceMap.toMap
}
